src/components/OrdersWidget/index.tsx

```diff
@@ -83,15 +88,41 @@ function classifyOrders(
   })
 }
 
-const compareFnFactory = (topic: TopicNames, asc: boolean) => (lhs: AuctionElement, rhs: AuctionElement): number => {
+function checkTokenAgainstSearch(token: TokenDetails | null, searchText: string): boolean {
+  if (!token) return false
+  return (
+    token?.symbol?.toLowerCase().includes(searchText) ||
+    token?.name?.toLowerCase().includes(searchText) ||
+    token?.address.toLowerCase().includes(searchText)
+  )
+}
+
+const filterOrdersFn = (searchTxt: string) => ({ id, buyToken, sellToken }: DetailedAuctionElement): boolean => {
+  if (searchTxt === '') return true
```

Comment:
Should this check be here as opposed to [useDataFilter](https://github.com/gnosis/dex-react/blob/ae8ab0c521/src/hooks/useDataFilter.tsx#L64-L65)?
By doing `data.filter(() => true)` we essentially clone array and mutate refrence.

Reply:
what if we need a custom `searchTxt` condition? can I use the built in 3rd `array` param of the filter function to use the reference somewhere?

Reply:
Also i don't really understand how that reference mutation happens, could you expand?

Reply:
> can I use the built in 3rd array
I don't see how

Reply:
if you need a `if(searchTxt.includes(*)` condition, then by all means include in `filterOrdersFn`
But as empty `searchTxt` is essentially `no filtering`, you can opt out early
```ts
const filterOrdersFn = (searchTxt: string) => {
  if (searchTxt === '') return null
  return ({ id, buyToken, sellToken }: DetailedAuctionElement): boolean => {
```
and then
```js
const customFilterFn = filterFnFactory(searchTxt)
if (!customFilterFn) return unfiltered_data
```

Reply:
> Also i don't really understand how that reference mutation happens, could you expand?
```js
const filteredData = useMemo(() => {
  return data.filter(() => true)
})
/// filteredData same as data, but
filteredData !== data


// optOut = true
const filteredData = useMemo(() => {
  return optOut ? data : data.filter(fn)
})
/// filteredData same as data and
filteredData === data
// reference stays the same
if we have useMemo(*, [filteredDate]) it won't rerurn
```

Reply:
thanks man